src/core/packages/ui-settings/server-internal/src/settings/index.ts

```diff
@@ -16,13 +16,14 @@ import { getThemeSettings } from './theme';
 import { getStateSettings } from './state';
 import { getAnnouncementsSettings } from './announcements';
 
-interface GetCoreSettingsOptions {
-  isDist?: boolean;
-  isThemeSwitcherEnabled?: boolean;
+export interface GetCoreSettingsOptions {
+  isServerless: boolean;
+  isDist: boolean;
+  isThemeSwitcherEnabled: boolean | undefined;
 }
 
 export const getCoreSettings = (
-  options?: GetCoreSettingsOptions
+  options: GetCoreSettingsOptions
```

Comment:
Is there a reason this was switched to required?

Reply:
It has only one usage, and I don't see why it would be optional if it stores important data like `isDist`. This change ensures nobody forgets to define it if this ever gets refactored.